types/leb128/.npmignore

```diff
@@ -0,0 +1,5 @@
+*
```

Comment:
Is this recommended? I'd prefer `files`.

Reply:
This is strictly required by the DefinitelyTyped infrastructure. All `d.ts` files are shipped, and no other files. An `npmignore` is used because there is no way in `files` to _exclude_ a versioned subdir if one exists.

Reply:
@jakebailey Won't `!subdir` work?

Reply:
I did a load of testing for this; `files` has different behavior between different package managers and even different versions. This isn't actually used by the DT publisher anyway (we copy all dts files), but this is mainly useful for those coming into the repo and running `npm pack` or `pacote` to get a close analog for a published package, which is useful for some tools.

Reply:
Both solutions are fine to me, I just  to point out that, if you think the behavior between package managers is inconsistent, I think you should raise a new issue for the un-common package manager instead, and the DT release workflow should be run with a certain Node version and known package manage already?

Reply:
DT releases are not performed using a package manager. We construct the packages directly to tarballs for security and reproducibility, then publish those. These files are here for external tools to get a reasonable representation of that process (as I described before), and so the format was chosen based on that use case.

I don't know what there is to discuss here; this file is required in this format and the PR will not pass without it, or with a `files` array in `package.json`. Changing that is definitely not the topic for this PR.

Reply:
I did in fact report an issue to yarn, for the inconsistency I found in `npmignore`. The files array was much more broken and did not behave how I expected, so I'm not as hopeful about that. Especially given the files array does not appear to actually work with negated patterns.

Reply:
> Changing that is definitely not the topic for this PR.

Indeed, I just saw this file and am a bit confused by this.

IMO, `files` is much easier to maintain based on my experience, but if it's the only solution you've ever tried then I have no problem here now.

Sorry to interrupt you. ❤️

Reply:
I use `files` myself! It's just not the right tool for this particular problem, which I wish weren't the case!

Reply:
Thanks @jakebailey for your patience and explanation. 🍻